Scripts/Runtime/Entities/PersistentData/Data/Interface/IEntityPersistentDataInstance.cs

```diff
@@ -0,0 +1,18 @@
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Represents an instance of data that is keyed off an associated <see cref="Entity"/>
+    /// for use in <see cref="IEntityPersistentData{T}"/> type.
+    /// </summary>
+    public interface IEntityPersistentDataInstance
+    {
+        /// <summary>
+        /// Called whenever an element is being disposed during cleanup.
+        /// This allows for custom cleanup code.
+        /// </summary>
+        /// <param name="entity">The entity that was associated with this data</param>
+        public void DisposeForEntity(Entity entity);
+    }
```

Comment:
Chatted about this already but since neither of us can come up with a use case for needing the Entity in disposal let's delete the current disposal method and have the interface extend `IDisposable`

Reply:
Done!